crates/core/component/dex/src/component/position_manager.rs

```diff
@@ -390,6 +392,10 @@ pub trait PositionManager: StateWrite + PositionRead {
             }
         }
 
+        // Update the candlestick tracking
+        self.record_position_execution(&prev_state, &new_state, &context)
+            .await?;
+
```

Comment:
Q: would it make sense to combine the candlestick recording with the event emission, either by moving this up to 377 where the event is emitted, or by moving the event emission _inside_ the candlestick call, so that we always record the candlestick data and the event together?

Reply:
I considered this however the event emission and candlestick creation are different concerns and the `position_execution` method is supposed to be the sole method tracking position executions and tying the various subsystems together so this seemed most design-legible to me.